static/src/javascripts/projects/common/modules/navigation/new-header.js

```diff
@@ -178,58 +177,63 @@ const toggleMenu = (): void => {
+const toggleDropdown = (menuAndTriggerEls: Object) => {
     const openClass = 'dropdown-menu--open';
 
-    fastdom
-        .read(() => ({
-            menu: document.querySelector('.js-edition-dropdown-menu'),
-            trigger: document.querySelector('.js-edition-picker-trigger'),
-        }))
-        .then(els => {
-            const { menu, trigger } = els;
+    fastdom.read(() => menuAndTriggerEls).then(els => {
+        const { menu, trigger } = els;
+
+        if (!menu) {
+            return;
+        }
 
-            if (!menu) {
-                return;
+        const isOpen = menu.classList.contains(openClass);
+        const expandedAttr = isOpen ? 'false' : 'true';
+        const hiddenAttr = isOpen ? 'true' : 'false';
+
+        return fastdom.write(() => {
+            if (trigger) {
+                trigger.setAttribute('aria-expanded', expandedAttr);
             }
 
-            const isOpen = menu.classList.contains(openClass);
-            const expandedAttr = isOpen ? 'false' : 'true';
-            const hiddenAttr = isOpen ? 'true' : 'false';
+            menu.setAttribute('aria-hidden', hiddenAttr);
+            menu.classList.toggle(openClass, !isOpen);
 
-            return fastdom.write(() => {
-                if (trigger) {
-                    trigger.setAttribute('aria-expanded', expandedAttr);
-                }
+            if (!isOpen) {
+                mediator.on('module:clickstream:click', clickSpec => {
+                    const elem = clickSpec ? clickSpec.target : null;
 
-                menu.setAttribute('aria-hidden', hiddenAttr);
-                menu.classList.toggle(openClass, !isOpen);
-
-                if (!isOpen) {
-                    mediator.on('module:clickstream:click', clickSpec => {
-                        const elem = clickSpec ? clickSpec.target : null;
-
-                        // if anywhere else but the links are clicked, the dropdown will close
-                        if (elem !== menu) {
-                            toggleEditionPickerDropdown();
-                        }
-                    });
-                } else {
-                    // when the dropdown closes, remove event
-                    mediator.removeEvent('module:clickstream:click');
-                }
-            });
+                    // if anywhere else but the links are clicked, the dropdown will close
+                    if (elem !== menu) {
+                        toggleDropdown(menuAndTriggerEls);
+                    }
+                });
+            } else {
+                // when the dropdown closes, remove event
+                mediator.removeEvent('module:clickstream:click');
+            }
+        });
+    });
+};
+
+const initiateUserAccountDropdown = () => {
+    const userAccountDropdownEls = {
+        menu: document.querySelector('.js-user-account-dropdown-menu'),
+        trigger: document.querySelector('.js-user-account-trigger'),
+    };
```

Comment:
Should we construct this object inside the `fastdom.read(...)` as well?

Reply:
Also, would it be possible to define a type for this? We could use it for the type of the param passed to `toggleDropdown(...)`